### PR TITLE
[script] Compare two build directories for number of bytecode instructions

### PR DESCRIPTION
**Pull request by vineethk**
## Description

In this PR, we add a simple python3 script to compare the number of bytecode instructions in two builds of the same projects.

To use this script, you need to first build some project with two different compiler configurations (such as different complier versions or turning some optimization off, etc). The you point the script at the two build directories.

## How Has This Been Tested?
Manually run when comparing the output of compiler v1 vs. v2 on some projects.

### CODE REVIEW

**Comment by vineethk** [comment]
* **#14315** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/14315?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @vineethk and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: third_party/move/scripts/compare_num_instructions.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+
+def count_instructions(file_name):
+    """
+    Count the number of bytecode instructions in the given file.
+    This requires disassembling the move bytecode.
+    """
+    result = subprocess.run(
+        [
+            "../../../target/debug/move-disassembler",
```
Comment (brmataptos) [inline third_party/move/scripts/compare_num_instructions.py:87]: Maybe there could be a more portable way to find the move-disassembler binary?

Perhaps `"cargo", "run", "-p", "move-disassembler", "--",`

**Review by brmataptos — APPROVED** [review]
Maybe could also read some env var for the cargo profile, e.g.:
```
   profile=os.environ.get('CARGO_PROFILE', "ci");
   ... subprocess.run(
       ...
       "--profile", profile,
       ...
```
I really think cargo should allow such an env var like that by default, but I can't find anything similar.

**Review by fEst1ck — APPROVED** [review]
Thanks!

**Pull request merged**